README.md

```diff
@@ -193,325 +243,482 @@ To make this happen, workers periodically phone home to Redis about each job the
+- `nearTermWindow`: number, ms; the window during which delayed jobs will be specifically scheduled using `setTimeout` - if all delayed jobs are further out that this window, the Queue will double-check that it hasn't missed any jobs after the window elapses.
+- `delayedDebounce`: number, ms; to avoid unnecessary churn for several jobs in short succession, the Queue may delay individual jobs by up to this amount.
+- `redis`: object or string, specifies how to connect to Redis. See [`redis.createClient()`](https://github.com/NodeRedis/node_redis#rediscreateclient) for the full set of options.
   - `host`: string, Redis host.
   - `port`: number, Redis port.
   - `socket`: string, Redis socket to be used instead of a host and port.
-  - `db`: number, Redis [DB index](http://redis.io/commands/SELECT).
-  - `options`: options object, passed to [node_redis](https://github.com/NodeRedis/node_redis#rediscreateclient).
-- `isWorker`: boolean, default true. Disable if this queue will not process jobs.
-- `getEvents`: boolean, default true. Disable if this queue does not need to receive job events.
-- `sendEvents`: boolean, default true. Disable if this worker does not need to send job events back to other queues.
-- `removeOnSuccess`: boolean, default false. Enable to have this worker automatically remove its successfully completed jobs from Redis, so as to keep memory usage down.
-- `catchExceptions`: boolean, default false. Only enable if you want exceptions thrown by the [handler](#queueprototypeprocessconcurrency-handlerjob-done) to be caught by Bee-Queue and interpreted as job failures. Communicating failures via `done(err)` is preferred.
+- `isWorker`: boolean. Disable if this queue will not process jobs.
+- `getEvents`: boolean. Disable if this queue does not need to receive job events.
+- `sendEvents`: boolean. Disable if this worker does not need to send job events back to other queues.
+- `storeJobs`: boolean. Disable if this worker does not need to associate events with specific `Job` instances.
+- `ensureScripts`: boolean. Ensure that the LUA scripts exist in redis before running any commands against redis.
+- `processDelayed`: boolean. Activate delayed jobs once they've passed their `delayUntil` timestamp.
+- `removeOnSuccess`: boolean. Enable to have this worker automatically remove its successfully completed jobs from Redis, so as to keep memory usage down.
+- `removeOnFailure`: boolean. Enable to have this worker automatically remove its failed jobs from Redis, so as to keep memory usage down. This will not remove jobs that are set to retry unless they fail all their retries.
+- `catchExceptions`: boolean. Only enable if you want exceptions thrown by the [handler](#queueprototypeprocessconcurrency-handlerjob-done) to be caught by Bee-Queue and interpreted as job failures. Communicating failures via `done(err)` is preferred.
```

Comment:
On the one hand, I feel like this has value for when you're misusing another library, and it throws a synchronous exception - then you get an `error` event instead of potentially retrying the job. On the other hand, `async` functions will reject the promise instead of throwing a synchronous exception. I think this makes sense, and folks will just need to deal with misconfigurations in some better manner.